### PR TITLE
[mysql] Fix dead loop when splitting chunks.

### PR DESCRIPTION
**Pull request by GOODBOY008**
Link issue https://github.com/ververica/flink-cdc-connectors/issues/1031.

### CODE REVIEW

File: flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/utils/ObjectUtils.java [hunk 1]
```diff
@@ -26,13 +26,21 @@ public class ObjectUtils {
...
     /**
      * Returns a number {@code Object} whose value is {@code (number + augend)}, Note: This method
-     * does not consider number overflow because we don't want to change the object type.
+     * will return max value of number type,if number overflows.
      */
     public static Object plus(Object number, int augend) {
         if (number instanceof Integer) {
-            return (int) number + augend;
+            try {
+                return Math.addExact((Integer) number, augend);
```
Comment (PatrickRen) [inline flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/utils/ObjectUtils.java:34]: What about throwing the `ArithmeticException` directly if the result overflows, and catch the exception in `ChunkSplitter`? Returning Integer.MAX or Long.MAX on overflow is a bit confusing and ambiguous.

**Review by PatrickRen — COMMENTED** [review]
Thanks for the patch @GOODBOY008 ! I left a comment

File: flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/utils/ObjectUtils.java (hunk 1 above)
Comment (GOODBOY008) [reply flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/utils/ObjectUtils.java:34]: Thks review,I will fix this.

File: flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/utils/ObjectUtils.java [hunk 2]
```diff
@@ -26,13 +26,13 @@ public class ObjectUtils {
 
     /**
      * Returns a number {@code Object} whose value is {@code (number + augend)}, Note: This method
-     * does not consider number overflow because we don't want to change the object type.
+     * will throw {@link ArithmeticException} if number overflows.
      */
     public static Object plus(Object number, int augend) {
```
Comment (PatrickRen) [inline flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/utils/ObjectUtils.java:31]: What about adding `ArithmeticException` to the method signature directly? Like:
```java
public static Object plus(Object number, int augend) throws ArithmeticException
```

File: flink-cdc-base/src/test/java/com/ververica/cdc/connectors/base/experimental/MySqlChunkSplitter.java [hunk 3]
```diff
@@ -220,7 +220,17 @@ private List<ChunkRange> splitEvenlySizedChunks(
...
+                chunkEnd = ObjectUtils.plus(chunkEnd, chunkSize);
+            } catch (ArithmeticException e) {
+                // Set chunkEnd to the max value of number type to avoid dead loop if number
+                // overflows.
+                chunkEnd = chunkEnd instanceof Integer ? Integer.MAX_VALUE : Long.MAX_VALUE;
+            }
+            // If chunkStart equals chunkEnd,should stop splitting chunk.
+            if (ObjectUtils.compare(chunkStart, chunkEnd) == 0) {
+                break;
+            }
```
Comment (PatrickRen) [inline flink-cdc-base/src/test/java/com/ververica/cdc/connectors/base/experimental/MySqlChunkSplitter.java:233]: Could we jump out of the loop here directly in the catch clause? Then all records greater than `chunkStart` (`chunkStart` to `MAX` in overflow case) will be treated as the ending split.

**Review by PatrickRen — COMMENTED** [review]
Thanks @GOODBOY008 for the update. I left some comments.

File: flink-cdc-base/src/test/java/com/ververica/cdc/connectors/base/experimental/MySqlChunkSplitter.java (hunk 3 above)
Comment (GOODBOY008) [reply flink-cdc-base/src/test/java/com/ververica/cdc/connectors/base/experimental/MySqlChunkSplitter.java]: @PatrickRen I have fixed.Thks for your tips.

**Review by ruanhang1993 — COMMENTED** [review]
Thanks for your work. I find something to fix.

**Comment by GOODBOY008** [comment]
@ruanhang1993 Thank you very much for your suggestions.I had add test and make method to public.

**Review by ruanhang1993 — APPROVED** [review]
LGTM. It is better to add another test which does not overflow.
